src/main/java/picard/analysis/TheoreticalSensitivity.java

```diff
@@ -204,4 +204,174 @@ public int draw() {
+                                                final int sampleSize, final double logOddsThreshold, final double alleleFraction) {
+        if (alleleFraction > 1.0 || alleleFraction < 0.0) {
+            throw new PicardException("Allele fractions must be between 0 and 1.");
+        }
+        double sensitivity = 0.0;
+        for (int k = 0; k < depthDistribution.length; k++) {
+            if (k % 100 == 0) {
+                log.info("Calculting sensitivity for allele fraction " + alleleFraction + " at depth " + k + " of " + depthDistribution.length);
+            }
+            sensitivity += sensitivityAtConstantDepth(k, qualityDistribution, logOddsThreshold, sampleSize, alleleFraction) * depthDistribution[k];
+        }
+        return sensitivity;
+    }
+
+    /**
+     * Removes trailing zeros in a distribution.  The purpose of this function is to prevent other
+     * functions from evaluating in regions where the distribution has zero probability.
+     * @param distribution Distribution of base qualities
+     * @return Distribution of base qualities removing any trailing zeros
+     */
+    public static double[] trimDistribution(final double[] distribution) {
+        int endOfDistribution = 0;
+
+        // Locate the index of the distribution where all the values remaining at
+        // larger indices are zero.
+        for (endOfDistribution = distribution.length-1;endOfDistribution >= 0;endOfDistribution--) {
+            if (distribution[endOfDistribution] != 0) {
+                break;
+            }
+        }
+
+        // Remove trailing zeros.
+        final double[] trimmedDistribution = new double[endOfDistribution+1];
+        for (int i = 0;i <= endOfDistribution;i++) {
+            trimmedDistribution[i] = distribution[i];
+        }
+
+        return trimmedDistribution;
+    }
+
+    /**
+     * This is a utility function
+     * @param theoreticalSensitivityOutput File to save to results ot theoretical sensitivity.
+     * @param tsOut MetricsFile object to save results of theoretical sensitivity to.
+     * @param sampleSize Number of samples to take for each depth.
+     * @param depthHistogram Histogram of depth distribution for sample.
+     * @param baseQHistogram Histogram of Phred-scaled quality scores.
+     * @param alleleFractions Allele fractions
+     */
+    public static void writeOutput(final File theoreticalSensitivityOutput, final MetricsFile<TheoreticalSensitivityMetrics, Double> tsOut, final int sampleSize,
+                                   final Histogram depthHistogram, final Histogram baseQHistogram, final List<Double> alleleFractions) {
+        if (theoreticalSensitivityOutput != null) {
+            final double logOddsThreshold = 6.2; // This threshold is used because it is the value used for MuTect2.
+            final double[] depthDoubleArray = TheoreticalSensitivity.normalizeHistogram(depthHistogram);
```

Comment:
`depthDistribution` or `depthPdf` would be a more descriptive name.

Reply:
fixed.